org.eclipse.jdt.core/batch/org/eclipse/jdt/internal/compiler/batch/ClasspathMultiReleaseJar.java

```diff
@@ -49,9 +49,13 @@ public void initialize() throws IOException {
 			} catch (FileSystemAlreadyExistsException e) {
 				this.fs = FileSystems.getFileSystem(uri);
 			}
-			this.releasePath = this.fs.getPath("/", "META-INF", "versions", this.compliance); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-			if (!Files.exists(this.releasePath)) {
-				this.releasePath = null;
+			for (int version = Integer.parseInt(this.compliance); version >= 9; version--) {
+				this.releasePath = this.fs.getPath("/", "META-INF", "versions", String.valueOf(version)); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
```

Comment:
according to
https://docs.oracle.com/javase/10/docs/specs/jar/jar.html#multi-release-jar-files
all versions down to 9 have to be checked.
`this.compliance` should be always a major version accroding to the spec, so `parseInt` should work

Reply:
Probably it would be good to parse the compliance only once (same below) and cache it in a field?